tty.go

```diff
@@ -28,6 +30,211 @@ func (t *tty) copyIO(w io.Writer, r io.ReadCloser) {
 	r.Close()
 }
 
+type EpollConsole struct {
```

Comment:
Also this needs a doc comment describing what it does, because I know the next time I'm going to have to touch this code it's going to be a fun time.